src/metatensor/models/experimental/pet/model.py

```diff
@@ -37,11 +38,16 @@ def __init__(
         super().__init__()
         self.name = ARCHITECTURE_NAME
         self.hypers = Hypers(hypers) if isinstance(hypers, dict) else hypers
-        self.cutoff = (
-            self.hypers["R_CUT"] if isinstance(self.hypers, dict) else self.hypers.R_CUT
-        )
+        self.cutoff = self.hypers.R_CUT
         self.all_species: List[int] = capabilities.species
         self.capabilities = capabilities
+        per_atom_output_types = [
+            output.per_atom for output in self.capabilities.outputs.values()
+        ]
+        if any(per_atom_output_types):
+            if not all(per_atom_output_types):
+                raise ValueError("All outputs must be per-atom or not per-atom.")
+            self.hypers.TARGET_TYPE = "atomic"
```

Comment:
The idea behind this check is that pet is usually initialized with a single `TARGET_TYPE` parameter, which means that in a single forward pass it can either predict per-atom or per-structure properties. If a mixture of per-atom and per-structure properties is requested in the output, then pet basically cannot work with this

Reply:
I understand that part of the check, my issue is that these `outputs` are re-used for the capabilities, while they should not be. Can PET be trained with `TARGET_TYPE="structure"` and then do inference with `TARGET_TYPE="atomic"`?

Reply:
Yes, I just realized that the best of way of implementing stuff is just have a default per-atom predictions with `TARGET_TYPE="atomic"`, and if non-per-atom property is requested in the output, then we just do sum over samples before returning the output values